app/components/op_primer/status_button_component.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module OpPrimer
+  class StatusButtonComponent < ApplicationComponent
+    include OpPrimer::ComponentHelpers
+
+    def initialize(current_status:, items:, readonly: false, button_arguments: {}, menu_arguments: {})
+      super
+
+      @current_status = current_status
+      @items = items
+      @readonly = readonly
+      @menu_arguments = menu_arguments
+      @button_arguments = button_arguments
+    end
+
+    def default_button_title
+      raise NotImplementedError
+    end
+
+    def disabled?
+      false
+    end
+
+    def readonly?
+      @readonly
+    end
+
+    def button_content(button)
+      button.with_leading_visual_icon(icon: @current_status.icon) if @current_status.icon
+      button.with_trailing_action_icon(icon: "triangle-down") if !readonly? && @items.any?
+
+      @current_status.name
+    end
+
+    def button_arguments
+      title = @button_arguments.fetch(:title) { default_button_title }
+
+      {
+        title:,
+        disabled: disabled?,
+        aria: {
+          label: title
+        },
+        style: @current_status.color&.color_styles_css
+      }.compact.deep_merge(@button_arguments)
+    end
+
+    def render_item(item)
+      flex_layout(align_items: :center) do |flex|
+        if item.icon
+          flex.with_column(mr: 1) do
+            render(Primer::Beta::Octicon.new(icon: item.icon))
+          end
+        elsif item.color
+          flex.with_column(mr: 1) do
+            helpers.icon_for_color(item.color)
+          end
+        end
+
+        flex.with_column do
+          render(Primer::Beta::Text.new) { item.name }
+        end
+      end
+    end
```

Comment:
Ideally, you'd call the `with_leading_visual_icon` slot on the menu item for the icon.